### DISCUSSION TITLE
Different regions for the same service

### DISCUSSION THREAD

**Discussion by ryno1234** [aws/aws-sdk-net#3225] [Q&A] [answered]
Hypothetically, if I wanted to use the `IAmazonS3` /  `AmazonS3Client`, but had different regions I needed to access, how would one go about doing that?

I know I could create a new instance of `AmazonS3Client` as needed, but I have to pass it credentials, i.e. `new AmazonS3Client(awsCredentials);`, which skates around the built in credential lookups specified here https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/creds-assign.html and requires me to know much more about how the credentials are configured (credentials file? IAM role? etc).

If I were to use the recommended way to configure the service, i.e. `services.AddAWSService<IAmazonS3>()`, I don't have control to configure multiple regions.

A solution here would be to allow some way to resolve `AWSCredentials` using the same lookup sequence mentioned in the link I just provided, however this doesn't seem to be exposed in the AWS .Net SDK. I thought the `AWSCredentialsFactory` would be what I was looking for, but turns out that has a different purpose.

**Comment by ashishdhingra** [comment, marked as answer]
@ryno1234 Good afternoon. If you use [new AmazonS3Client()](https://github.com/aws/aws-sdk-net/blob/f3b10709a97017b6b1dcc3e7275e4e455526885d/sdk/src/Services/S3/Generated/_netstandard/AmazonS3Client.cs#L64) or [new AmazonS3Client(RegionEndpoint)](https://github.com/aws/aws-sdk-net/blob/f3b10709a97017b6b1dcc3e7275e4e455526885d/sdk/src/Services/S3/Generated/_netstandard/AmazonS3Client.cs#L83C31-L83C45) without credentials, the SDK would try to resolve the credentials automatically using the resolution order mentioned in [Credential and profile resolution](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/creds-assign.html). If you would like to manually resolve the credentials using the same order, you could use [Amazon.Runtime.FallbackCredentialsFactory.GetCredentials()](https://github.com/aws/aws-sdk-net/blob/f3b10709a97017b6b1dcc3e7275e4e455526885d/sdk/src/Core/Amazon.Runtime/Credentials/FallbackCredentialsFactory.cs#L129C38-L129C52) (although this manual step is not required since `new AmazonS3Client(RegionEndpoint)` would have credential resolution done automatically). Please advise if this answers your question.

Thanks,
Ashish

**Reply by ryno1234** [reply]
Thank you
